src/view/sort-view.js

```diff
@@ -1,38 +1,34 @@
 import AbstractView from './abstract-view.js';
+import {SortType} from '../constants.js';
+
+const createMainSortTemplate = (sortType) => {
+  const sortDirections = Object.values(SortType);
+  return `<form class="trip-events__trip-sort  trip-sort" action="#" method="get">
+    ${sortDirections.map(({name, disabled}) => `<div class="trip-sort__item  trip-sort__item--${name}">
+    <input id="sort-${name}" class="trip-sort__input  visually-hidden" data-sort-type="${name}" type="radio" name="trip-sort" value="sort-${name}" ${sortType === name ? 'checked' : ''} ${disabled === true ? 'disabled' : ''}>
+    <label class="trip-sort__btn" for="sort-${name}">${name}</label>
+  </div>`).join('')}
+  </form>`;};
 
-const createMainSortTemplate = () => (
-  `<form class="trip-events__trip-sort  trip-sort" action="#" method="get">
-  <div class="trip-sort__item  trip-sort__item--day">
-    <input id="sort-day" class="trip-sort__input  visually-hidden" type="radio" name="trip-sort" value="sort-day">
-    <label class="trip-sort__btn" for="sort-day">Day</label>
-  </div>
-
-  <div class="trip-sort__item  trip-sort__item--event">
-    <input id="sort-event" class="trip-sort__input  visually-hidden" type="radio" name="trip-sort" value="sort-event" disabled>
-    <label class="trip-sort__btn" for="sort-event">Event</label>
-  </div>
-
-  <div class="trip-sort__item  trip-sort__item--time">
-    <input id="sort-time" class="trip-sort__input  visually-hidden" type="radio" name="trip-sort" value="sort-time">
-    <label class="trip-sort__btn" for="sort-time">Time</label>
-  </div>
-
-  <div class="trip-sort__item  trip-sort__item--price">
-    <input id="sort-price" class="trip-sort__input  visually-hidden" type="radio" name="trip-sort" value="sort-price" checked>
-    <label class="trip-sort__btn" for="sort-price">Price</label>
-  </div>
-
-  <div class="trip-sort__item  trip-sort__item--offer">
-    <input id="sort-offer" class="trip-sort__input  visually-hidden" type="radio" name="trip-sort" value="sort-offer" disabled>
-    <label class="trip-sort__btn" for="sort-offer">Offers</label>
-  </div>
-</form>`
-);
 
 class SortView extends AbstractView {
   get template() {
     return createMainSortTemplate();
   }
+
+  setSortTypeChangeHandler = (callback) => {
+    this._callback.sortTypeChange = callback;
+    this.element.addEventListener('click', this.#sortTypeChangeHandler);
+  }
+
+  #sortTypeChangeHandler = (evt) => {
+    if (evt.target.tagName !== 'INPUT') {
+      return;
+    }
+
+    evt.preventDefault();
+    this._callback.sortTypeChange(evt.target.dataset.sortType);
```

Comment:
```suggestion
    const { sortType  } = evt.target.datase;
    const hasAttr = Boolean(sortType);
    if (!hasAttr) {
      return;
    }

    evt.preventDefault();
    this._callback.sortTypeChange(evt.target.dataset.sortType);
```
лучше и проверять на наличие атрибута